src/main/java/com/marklogic/client/dataloader/impl/DefaultSchemasDataLoader.java

```diff
@@ -0,0 +1,47 @@
+package com.marklogic.client.dataloader.impl;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.dataloader.SchemasDataFinder;
+import com.marklogic.client.dataloader.SchemasDataLoader;
+import com.marklogic.client.document.TextDocumentManager;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.helper.LoggingObject;
+import com.marklogic.client.io.FileHandle;
+import com.marklogic.client.io.Format;
+
+public class DefaultSchemasDataLoader extends LoggingObject implements SchemasDataLoader {
+	
+	@Override
+	public Set<File> loadSchemasData(File baseDir, SchemasDataFinder schemasDataFinder, DatabaseClient client) {
+		XMLDocumentManager xmlDocMgr = client.newXMLDocumentManager();
+		TextDocumentManager textDocMgr = client.newTextDocumentManager();
+		
+		List<File> schemasData = schemasDataFinder.findSchemasData(baseDir);
+
+        Set<File> loadedSchemasData = new HashSet<>();
```

Comment:
Don't forget to add the added files to this Set. IIRC, I did this for modules so that in a JUnit environment, the test infrastructure could look at the modules that had been added and then conditionally perform some more logic before the test runs. I don't know of a similar use case yet for schemas (I've never used schemas), but I think getting the Set back is easy and could be helpful in the future.